src/main.c

```diff
@@ -3026,6 +3077,35 @@ static void on_accept(h2o_socket_t *listener, const char *err)
     } while (--num_accepts != 0);
 }
 
+static void on_h2olog_accept(h2o_socket_t *listener, const char *err)
+{
+    if (err != NULL) {
+        return;
+    }
+
+    h2o_socket_t *sock;
+    if ((sock = h2o_evloop_socket_accept(listener)) == NULL)
+        return;
+
+    h2o_socket_export_t sockinfo;
+    if (h2o_socket_export(sock, &sockinfo) != 0) {
+        h2o_error_printf("failed to export h2o socket\n");
+        return;
+    }
```

Comment:
Ah right, we can call `cloexec_accept` here for simplicity?

Reply:
I see.

My understanding is that `h2o_evloop_socket_accept` does `cloexec_accept` and setup some other stuff for `h2o_socket_t`, but `h2o_socket_export` extracts a raw fd and cleanups the other stuff, so if one calls `h2o_socket_export` immediately after `h2o_evloop_socket_accept`, calling `cloexec_accept` directly is good enough.